src/recorder.h

```diff
@@ -46,16 +46,20 @@ class Recorder {
   friend class Datum;
 
  public:
-  /// create a new recorder with default dump frequency and random
-  /// simulation id.
+  /// create a new recorder with default dump frequency, random
+  /// simulation id, and simulation id injection.
   Recorder();
 
+  /// create a new recorder with the given dump count, random
+  /// simulation id, and given flag for injecting the simulation id.
+  Recorder(bool inject_sim_id);
+
   /// create a new recorder with the given dump count and random
   /// simulation id.
   Recorder(unsigned int dump_count);
 
-  /// create a new recorder with default dump frequency and the specified
-  /// simulation id.
+  /// create a new recorder with default dump frequency. the specified
+  /// simulation id, and simulation id injection.
   Recorder(boost::uuids::uuid simid);
```

Comment:
Then, at minimum, I would suggest a single constructor with all possible
(non default) args.

On Sun, Jan 4, 2015 at 9:01 PM, Anthony Scopatz <notifications@github.com>
wrote:

> In src/recorder.h
> <https://github.com/cyclus/cyclus/pull/1062#discussion-diff-22446380>:
>
> >    /// create a new recorder with the given dump count and random
> >    /// simulation id.
> >    Recorder(unsigned int dump_count);
> >
> > -  /// create a new recorder with default dump frequency and the specified
> > -  /// simulation id.
> > +  /// create a new recorder with default dump frequency. the specified
> > +  /// simulation id, and simulation id injection.
> >    Recorder(boost::uuids::uuid simid);
>
> Good to know that they are allowed. However, it is allowed written and
> consistent with what was there before. If more constructors are not
> blocking then I can't do this. I am effectively out of time for this and
> for any nice-to-have features. I need to finish writing skeleton cymetric
> docs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1062/files#r22446380>.
>



-- 
Matthew Gidden
Ph.D. Candidate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192